orderbook/src/trade_events.rs

```diff
@@ -57,13 +57,13 @@ impl TradeEvents {
             let trades = trades.context("failed to get event")?;
             if !have_deleted_old_events {
                 self.db
-                    .replace_trades(*range.start(), &[])
+                    .replace_trades(*range.start(), trades)
```

Comment:
Out of curiosity is there a reason you want the deletion and first insertion atomically if the other insertions are not done atomically?

Reply:
This line was a bug in the previous PR. It should have inserted the trades all along. There is a long comment a couple of lines above that explains this reasoing.